### ISSUE TITLE
Reading Excel File is not working with IE 11

### ISSUE THREAD

**Issue by gnachao** [agershun/alasql#397]
It's not working with IE 11! 

http://alasql.org/demo/008file/

**Comment by mathiasrw** [comment]
Thanks for letting us know. I Will look into this

**Comment by mathiasrw** [comment]
@agershun http://alasql.org/demo/008file/ is using 0.1.9 - Could you please update the html to include 

    <script src="//cdn.jsdelivr.net/alasql/latest/alasql.min.js"></script>

**Comment by gnachao** [comment]
It's for sure NOT working for IE 11.

**Comment by mathiasrw** [comment]
What are you experiencing?

Do you get any error in the console?

What file do you use?

**Comment by gnachao** [comment]
Sorry for not being specific. The demo code bellow addresses what work and what doesn't.

```
<script src="https://cdn.jsdelivr.net/alasql/latest/alasql.min.js"></script>
<script src="http://alasql.org/console/xlsx.core.min.js"></script>
<script src="http://alasql.org/console/Blob.js"></script>

<h1>File load and save</h1>
<p>Upload file here</p>
<input id="readfile" type="file" onchange="loadFile(event)"/>
<p>Save file here:</p>
<button onclick="saveFile()">Save file</button>
<div id="res"></div>
<xmp id="result"></xmp>

<script>
	var data;

	// WORK all except IE 11
	function loadFile(event) {
		alasql('SELECT * FROM FILE(?,{headers:true})',[event],function(res){
			data = res;
			document.getElementById("res").textContent = JSON.stringify(res);
		});
	}

	// WORK all except IE 11
	function saveFile() {
		console.log('saveFile');
		alasql('SELECT * INTO XLSX("myfile.xlsx",{headers:true}) FROM ?',[data]);
	}

	//WORK: Safari, Firefox; NOT WORK: Chrome, IE 11
	alasql('select * from xlsx("cities.xlsx", {headers:true, sheetid:"Sheet1", range:"A1:B6"})',[],function(data){
		document.getElementById('result').textContent = JSON.stringify(data);
	});

</script>
```

**Comment by agershun** [comment]
@mathiasrw Just changed all demos to CDN. Could you verify it?

**Comment by gnachao** [comment]
```javascript
var data;
alasql('select * from xlsx("cities.xlsx", {headers:true})',[],function(res){
   data = res;
});
```
The above code will not work with Chrome or IE. Is there any temporary solution just to open xlsx and pass to data?

Thank you!

**Comment by mathiasrw** [comment]
@agershun Will verify tonight

@gnachao 

Thank you for a really good description. In this very moment I dont have access to IE 11 (only tonight) but will play with your example as soon as I get home. 

I must ask you one thing: are you running this from a flat `.html` file or via a server?

**Comment by gnachao** [comment]
Thank you for your reply. My code is running on a local flat .html.

**Comment by mathiasrw** [comment]
@gnachao 

Makes sense then. 

Please try to load from a http server - if you have python installed just run `python -m SimpleHTTPServer` - you might have apache installed already - or run tests at your web host. 

Put the .xlsx file in the same folder at your .html (or change to absolute path as for example from `"cities.xlsx"` to `"http://localhost/testing/cities.xlsx"`)

**Comment by gnachao** [comment]
@mathiasrw 

Thank you for suggestion. 

YES, the ```cities.xlsx``` is in the same folder with .html file. (if it wasn't, it would be not working completly but I did mention in the code that it work in safari and firefox )

On a local machine, there is a project folder containing index.html and cities.xlsx. Does the following code works without any further configuration or installation? (works means data store content of cities.xlsx)
```javascript
var data;
alasql('select * from xlsx("cities.xlsx", {headers:true})',[],function(res){
   data = res;
});
```  

The sample code posted 5 days ago is complete, If you try with having the cities.xlsx in the folder, you will see the effect I commented in the code.

My project is running on a local machine only not on a sever, and I thought alasql work with local machine as the author mentioned in couple places including in stackoverflow. Reading .xlsx to JS Object is the most important feature and I do think it should  working.

**Comment by gnachao** [comment]
Even if you change to the latest CDN of the following link, it still now working with IE.

http://alasql.org/demo/008file/

**Comment by mathiasrw** [comment]
I will look into the demo issue - but if you run things on a local machine I really recommend using node and getting alasql via npm

**Comment by gnachao** [comment]
A folder contains:
   cities.xlsx
   index.html

```javascript
<script src="https://cdn.jsdelivr.net/alasql/latest/alasql.min.js"></script>
<script src="http://alasql.org/console/xlsx.core.min.js"></script>
<scirpt>
var data;
alasql('select * from xlsx("cities.xlsx", {headers:true})',[],function(res){
   data = res;
});
</script>
```
@mathiasrw  Do you mean the above code is not working from the browser? Could you please give me a bit more info. of how to use node and getting alasql via npm?

**Comment by mathiasrw** [comment]
you computer limits your access to files on your filesystem.

Thats why the code is correct, but accessing the file "cities.xlsx" depends on the browsers safty features. 

Please see if you can serve this with a local webserver. Checkout www.uwamp.com if you are on windows

**Comment by gnachao** [comment]
If that is the case; then, could you please make this demo http://alasql.org/demo/008file/ work with IE 11? Thank you!

**Comment by mathiasrw** [comment]
I will try

**Comment by salilvnair** [comment]
If anyone is facing this Issue here is the update: issue is with **readAsBinaryString **which is deprecated in IE, so replace it with **readAsArrayBuffer **in alasql.min.js and it should work.
tested with IE ,FF working fine.

**Comment by mathiasrw** [comment]
@salilvnair SO awesome you just spotted this. 

Can someone with IE11 please verify that the last develop version https://github.com/agershun/alasql/blob/develop/dist/alasql.min.js fixes the problem?

**Comment by mathiasrw** [comment]
Closed per request from #628
